src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -123,6 +129,17 @@ public async Task ExecuteAsync(ExportJobRecord exportJobRecord, WeakETag weakETa
                 // Connect to export destination using appropriate client.
                 await _exportDestinationClient.ConnectAsync(exportJobConfiguration, cancellationToken, _exportJobRecord.StorageAccountContainerName);
 
+                // Add a request context so that bundle issues can be added by the SearchOptionFactory
+                var fhirRequestContext = new FhirRequestContext(
+                method: "Export",
+                uriString: "$export",
+                baseUriString: "$export",
+                correlationId: _exportJobRecord.Id,
+                requestHeaders: new Dictionary<string, StringValues>(),
+                responseHeaders: new Dictionary<string, StringValues>());
+
+                _contextAccessor.FhirRequestContext = fhirRequestContext;
```

Comment:
Should we only set this if its null?

Reply:
It has always been null in my testing. I'm concerned about adding a null check as this works on the assumption that the request context doesn't matter outside of recording issues.

Reply:
My general rule of thumb/concern is mutating state on an object thats outside of the creating or owning component. In this case _contextAccessor is not set because it's outside of the http pipeline, but the point being ExportJobTask has no visibility if other components require this component in this scope. I'd say to keep the assumptions of other components (that you can't see), the right pattern here is to save the existing value, set your new one, then set it back after.